rig-core/rig-core-derive/src/lib.rs

```diff
@@ -288,7 +288,7 @@ pub fn rig_tool(args: TokenStream, input: TokenStream) -> TokenStream {
 
     let expanded = quote! {
         #[derive(serde::Deserialize)]
-        struct #params_struct_name {
+        pub(crate) struct #params_struct_name {
```

Comment:
Does this struct need to be visible outside the crate. when a side-crate is using this macro, it'll create a struct that's visible within the crate it's invoked in and not globally which I think makes sense? but it might be awkward if u want to use that struct in like a side-crate if u have a multi-crate repo.

I think it's fine to do this, we can always make it more public later but i wanted to think about it.

Reply:
yea, figured! lets send it!